### DISCUSSION TITLE
migu的cookie中并没有aversionid这个key

### DISCUSSION THREAD

**Discussion by Syozhi-1** [1715173329/UnblockNeteaseMusic#549] [Q&A]
登陆migu音乐官网，在控制台中输入 document.cookie 得到的cookie中并没有aversionid这个key，具体key如下图所示，请问是这个id更换了key名，还是取其中的哪个值，望告知，谢谢
![image](https://user-images.githubusercontent.com/58074198/155716251-b0a68c8f-e27b-44b2-ba96-e5095a869c45.png)

**Comment by 1715173329** [comment]
下载咪咕音乐手机 app（最好是 Android 端），登录并抓包。

**Reply by Syozhi-1** [reply]
我尝试了使用VENT进行抓包，得到几个看起来应该是的值，
aid，msisdn，userId，sessionId，signature
其中aid是在进行登录时抓到的，这几个值我都试了一遍，都无法在ypm中播放周杰伦的歌曲，ypm也没有报错；
也许抓包抓到正确的值了，但是我没有看到aversionid这个key；
我猜aid可能是，或许我的使用方式也有问题，我在windows的环境变量中添加了MIGU_COOKIE，但就是无法播放周董的歌，求解

**Reply by 1715173329** [reply]
aversionid 由 168 位纯大写字母和数字构成，请确认

**Comment by SenLief** [comment]
我找到了这个aversionid，但是没什么用，好像migu的音源已经挂了，migu一直报错url错误。

**Reply by debugbb** [reply]
ios抓包，看到个bversionid，没看到有aversionid，只能在android上抓到aversionid么？谢谢

**Comment by SenLief** [comment]
在安卓上可以看到。

发自我的iPhone

在 2022年8月26日，13:59，debugbb ***@***.***> 写道：

﻿

ios抓包，看到个bversionid，没看到有aversionid，只能在android上抓到aversionid么？谢谢

**Reply by debugbb** [reply]
好的，我等着找个安卓机试试，谢谢

**Comment by debugbb** [comment]
经测试，安卓机抓出来的是aversionid，ios抓出来的是bversionid，这俩都能用

**Comment by qu270051387** [comment]
@1715173329 在23年的今天我也碰到了这个问题  我使用小黄鸟在手机上抓到了咪咕APP的aversionid 但是有很多很多个  而且结果都不一样 由纯大写字母和数字构成  但是不是168位

**Reply by 1715173329** [reply]
现在是 166 位，不一样是正常的，选一个即可。
这当中包含你的账号信息，所以长期有效。

**Reply by qu270051387** [reply]
不啊  我现在抓到的结果都是104位  相差甚远

**Reply by qu270051387** [reply]
ENABLE_FLAC=true MIGU_COOKIE="DFDFD0D3DDA0A78F6A978FD2B57B9C78C99A878D92ECEED4639A899B857F9F6F9491858792ECA38E6B9187A18B819A749A9984" node app.js -o migu -p 5000:5002 -e http://music.163.com -s   我是这么运行的  不知道migucookie是否生效 因为我是有咪咕会员的  按照我的理解 这种情况下应该就能听周杰伦的FLAC了  但是目前还是128k

**Reply by 1715173329** [reply]
用 Android 7.1.6 版本，或者你也可以抓下新的播放 api

**Reply by qu270051387** [reply]
嗯   我重新抓取了一下aversionid 196个字节  现在migu源可以播放flac了  但是会有匹配不准的情况  我打算添加其他的源+SEARCH_ALBUM=true来进行限制  但是有个问题 现在QQ是不是无法获取高品质了即便有绿钻  我打算使用ytdlp 但是服务器无法访问youtube 还有就是  我现在的服务器IP和端口添加在杜比大喇叭里  使用代理服务器模式的时候  无法正常解锁  提示播放地址失败  该如何解决

**Reply by qu270051387** [reply]
@1715173329 
ENABLE_FLAC=true FOLLOW_SOURCE_ORDER=true MIN_BR=320000 MIGU_COOKIE="************************" node app.js -o migu kuwo bilibili -p 5000:5002 -e http://music.163.com -s
这是现在的规则  有几个问题
1.云盘里的歌曲无法播放  
2.网易云本身有版权的歌曲无法播放 比如[老虎歌皇]
3.匹配错误  我的想法是优先咪咕 因为能获取FLAC 其次是kuwo bilibili等   但是现在虽然优先咪咕了 可是匹配的不一定准确  会出现播放的是FLAC 但是歌是错的  如果加入专辑匹配变量  又不能正确匹配咪咕的FLAC 会出现低音质或者匹配错误  比如夜曲
4.杜比大喇叭使用代理服务器模式  无法正常解锁 提示播放地址失败 PC端可以解锁  安卓端通过修改wifi代理可以解锁
